docs/proposals/scheduling/binding-priority-preemption/README.md

```diff
@@ -0,0 +1,340 @@
+By doing so, they can achieve the same effect as updating a priority class, but we still do not allow updating priority classes to prevent accidental changes.
+
+### Preemption scenario
+
+In this proposal, the only scenario under which a group of bindings in Karmada may be preempted is when a higher priority binding cannot be scheduled due to various unmet scheduling requirements,
+such as lack of resources, spread constraints, etc., and the preemption of the lower priority bindings allows the higher priority binding to be scheduled.
+So, if the preemption of the lower priority bindings does not help with scheduling of the higher priority binding, those lower priority bindings will keep running and the higher priority binding will stay pending.
+
+**Important notes**: The scheduler will only consider bindings whose resource requirements are not empty.
+From scheduler's point of view, such a binding needs no resources and its preemption will not release any resources.
+
+### Preemption order
+
+When scheduling a pending binding, scheduler tries to place the binding on a member cluster that does not require preemption.
+If there is no such a member cluster, scheduler may favor a member cluster where the number and/or priority of victims (preempted bindings) is smallest.
+After choosing the member cluster, scheduler considers the lowest priority bindings for preemption first.
+Scheduler starts from the lowest priority and considers enough bindings that should be preempted to allow the pending binding to schedule.
+Scheduler only considers bindings that have lower priority than the pending binding.
+
+Scheduler will try to minimize the number of preempted bindings.
+As a result, it may preempt a binding while leaving lower priority bindings running if preemption of those lower priority bindings is not enough to schedule the pending binding while preemption of the higher priority binding(s) is enough to schedule the pending binding.
+For example, if member cluster capacity is 10, and pending binding is priority 10 and requires 5 units of resource, and the running bindings are
+```json lines
+{
+    "priority": 0,
+    "request": 1
+}
+{
+    "priority": 1,
+    "request": 2
+}
+{
+    "priority": 2,
+    "request": 5
+}
+{
+    "priority": 3,
+    "request": 2
+}
+```
+, scheduler will preempt the priority 2 binding only and leaves priority 1 and priority 0 running.
+
+### Preemption in multi-scheduler
+
+Karmada allows multiple schedulers to exist in its control plane, This introduces a race condition where multiple schedulers may perform round-robin preemption.
+That is, scheduler A may schedule binding A, Scheduler B preempts binding A to schedule binding B which is then preempted by scheduler A to schedule binding A and we go in a loop.
+
+Therefore, we strongly recommend not to enable preemption behavior in multiple schedulers.
+If you must enable it, unless you can ensure that they will not have any intersection during preemption.
```

Comment:
Good point!
I think each scheduler can only preempt ResourceBinding managed by itself.

Reply:
yes, that’s the point.